src/Microsoft.Kiota.Cli.Commons/IO/TableOutputFormatter.cs

```diff
@@ -100,78 +100,55 @@ public TableOutputFormatter(IAnsiConsole console)
         }
         else
         {
-            var restrictedValueKinds = new JsonValueKind[] {
-                    JsonValueKind.Array,
-                    JsonValueKind.Object
-                };
-            var objectEnumerator = firstElement.EnumerateObject();
-            var buffer = new List<string>();
-            foreach (var property in objectEnumerator)
+            foreach (var property in firstElement.EnumerateObject())
             {
-                if (restrictedValueKinds.Contains(property.Value.ValueKind)) {
-                    continue;
-                }
-
-                buffer.Add(property.Name);
+                if (property.Value.ValueKind is JsonValueKind.Array or JsonValueKind.Object) continue;
+                yield return property.Name;
             }
-            propertyNames = buffer;
         }
-        
-        return propertyNames;
     }
 
     private static IEnumerable<IRenderable> GetRowColumns(IEnumerable<string> propertyNames, JsonElement row)
     {
-        return propertyNames.Select(p =>
+        foreach (var columnName in propertyNames)
         {
-            var propertyName = p;
-            if (row.ValueKind == JsonValueKind.Object)
-            {
-                var hasProp = row.TryGetProperty(propertyName, out var property);
-                if (hasProp)
-                    return GetPropertyValue(property);
-                else
-                    return new Markup("-");
-            }
-
-            return GetPropertyValue(row);
-        });
+            yield return RenderCell(columnName, row);
+        }
+    }
+    
+    private static Markup RenderCell(string columnName, JsonElement row)
+    {
+        if (row.ValueKind != JsonValueKind.Object) return GetPropertyValue(row);
+        var hasProp = row.TryGetProperty(columnName, out var property);
+        return hasProp ? GetPropertyValue(property) : new Markup(CellValueIfMissing);
     }
 
-    private static IRenderable GetPropertyValue(JsonElement property)
+    private static Markup GetPropertyValue(JsonElement property)
     {
         var valueKind = property.ValueKind;
-        object? value = null;
-        switch (valueKind)
+        return valueKind switch
         {
-            case JsonValueKind.String:
-                value = property.GetString();
-                break;
-            case JsonValueKind.True:
-            case JsonValueKind.False:
-                value = property.GetBoolean();
-                break;
-            case JsonValueKind.Number:
-                value = property.GetDecimal();
-                break;
-        }
-        return new Markup(value?.ToString() ?? "-");
+            JsonValueKind.String => new Markup(property.GetString() ?? CellValueIfMissing),
+            JsonValueKind.True or JsonValueKind.False => new Markup(property.GetBoolean().ToString()),
+            JsonValueKind.Number => new Markup(property.GetDecimal().ToString(CultureInfo.CurrentCulture)),
```

Comment:
```suggestion
            JsonValueKind.Number => new Markup(property.GetDecimal().ToString(CultureInfo.InvariantCulture)),
```

I'm not sure it makes sense to localize the number formats for a CLI?

Reply:
The end user sees it. That's why I thought to use their locale. For example, some people use commas as decimal separators.

Reply:
If the goal is to help the user read numbers in their preferred culture, shouldn't this use the CurrentUICulture?
Also, what about right to left cultures? is the console flipping around? 
We need to consider the output might be mechanically parsed by something else, and having the output localized would break that.

Reply:
We can use the invariant culture and collect user feedback.